components/org.wso2.carbon.identity.local.auth.smsotp.event.handler/src/main/java/org/wso2/carbon/identity/local/auth/smsotp/event/handler/notification/internal/SMSNotificationUtil.java

```diff
@@ -0,0 +1,159 @@
+        StringBuilder smsBody = new StringBuilder();
+        while (placeHolderMatcher.find()) {
+            String placeholder = placeHolderMatcher.group(1);
+            String replacement = filteredPlaceholderData.get(placeholder);
+            if (StringUtils.isBlank(replacement)) {
+                continue;
+            }
+            placeHolderMatcher.appendReplacement(smsBody, Matcher.quoteReplacement(replacement));
+        }
+        placeHolderMatcher.appendTail(smsBody);
+        return smsBody.toString();
+    }
+
+    /**
+     * Resolve OTP values in the event properties and add them to the notification data.
+     *
+     * @param dataMap Notification data.
+     */
+    public static void resolveOTPValues(Map<String, Object> dataMap, Map<String, String> notificationData) {
+
+        OTP otp = (OTP) dataMap.get(SMSNotificationConstants.OTP_TOKEN_PROPERTY_NAME);
+        if (otp != null) {
+            notificationData.put(SMSNotificationConstants.PLACE_HOLDER_CONFIRMATION_CODE, otp.getValue());
+            notificationData.put(SMSNotificationConstants.PLACE_HOLDER_OTP_EXPIRY_TIME,
+                    String.valueOf(TimeUnit.MILLISECONDS.toMinutes(otp.getValidityPeriodInMillis())));
+        } else {
+            String otpCode = (String) dataMap.get(OTP_TOKEN_STRING_PROPERTY_NAME);
+            if (StringUtils.isNotBlank(otpCode)) {
+                notificationData.put(SMSNotificationConstants.PLACE_HOLDER_CONFIRMATION_CODE, otpCode);
+            }
+        }
+    }
+
+    /**
+     * Get if applying SMS templates is enabled.
+     *
+     * @return boolean if applying SMS templates is enabled.
+     */
+    public static boolean isEnabledSMSTemplates() {
+
+        return Boolean.parseBoolean(IdentityUtil.getProperty(SMSNotificationConstants.PROPERTY_APPLY_SMS_TEMPLATES));
+    }
+
+    /**
+     * Filter out the placeholders that are not accepted in the SMS template.
+     *
+     * @param notificationData Notification data.
+     * @return Filtered notification data.
+     */
+    public static Map<String, String> filterPlaceHolderData(Map<String, String> notificationData) {
+
+        Map<String, String> result = new HashMap<>();
+        for (Map.Entry<String, String> entry : notificationData.entrySet()) {
+            if (ACCEPTED_SMS_PLACEHOLDERS.contains(entry.getKey())) {
+                result.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return result;
+    }
+
+    /**
+     * If the tenant domain is a UUID, resolve the organization name from the associated organization resource.
+     *
+     * @param tenantDomain Tenant domain.
+     * @return Human-readable name related to the represented organization space.
+     * @throws IdentityEventException Error while resolving organization name.
+     */
+    public static String resolveHumanReadableOrganizationName(String tenantDomain) throws IdentityEventException {
+
+        String organizationName = tenantDomain;
```

Comment:
What could be the reason for this reassignment?

Reply:
if it's super tenant, or resources needed to resolve the organizationName is not resolved, the tenantDomain is returned as the organization name.